services/gossip/adapter/tcp/server.go

```diff
@@ -12,56 +12,91 @@ import (
 	"github.com/orbs-network/govnr"
 	"github.com/orbs-network/membuffers/go"
 	"github.com/orbs-network/orbs-network-go/instrumentation/logfields"
+	"github.com/orbs-network/orbs-network-go/instrumentation/metric"
 	"github.com/orbs-network/orbs-network-go/instrumentation/trace"
 	"github.com/orbs-network/orbs-network-go/services/gossip/adapter"
 	"github.com/orbs-network/scribe/log"
 	"github.com/pkg/errors"
 	"net"
+	"sync"
 	"time"
 )
 
-func (t *DirectTransport) serverListenForIncomingConnections(ctx context.Context, listenPort uint16) (net.Listener, error) {
+type serverConfig interface {
+	GossipListenPort() uint16
+	GossipNetworkTimeout() time.Duration
+}
+
+type transportServer struct {
+	sync.RWMutex
```

Comment:
ideally the mutex (along with `lock()`, `unlock()`) should be a private implementation detail of the server.

Reply:
It is. Embedding a mutex is the idiomatic practice.